:memo: Update Docker/Podman related documentation (#624)
it-at-m/refarch: docs/templates/develop.md

Key technologies used in the templates include:

### Container Engine

> [Podman](https://podman.io) is suggested to run a local development stack including all necessary services. Alternatively [Docker](https://www.docker.com/)

Reviewer: ```suggestion
[Podman](https://podman.io) is suggested for running the local development stack including all necessary services. Alternatively [Docker](https://www.docker.com/)
```
